parsers/manifest_parser.go

```diff
@@ -274,6 +274,9 @@ func (dm *YAMLParser) ComposeActions(mani *ManifestYAML, manipath string) (ar []
 			wskaction.Exec.Kind = action.Runtime
 		}
 
+                // we can specify the name of the action entry point using main
+                wskaction.Exec.Main = action.Main
```

Comment:
Here the main function is missing from the action file and therefore we are using "--main" parameter of `wsk` CLI to designate other function as main. Also, not sure how we can produce an error message from wskdeploy when main function is missing from the action file and `--main` parameter is not populated. `wsk` CLI has this information and does error out saying "'main' must be a string defined in 'exec' for 'java' actions".

Let me know if I am missing anything. Happy to fix it.